CMakeLists.txt

```diff
@@ -254,14 +255,39 @@ set_target_properties(dppp_testdyndppp PROPERTIES VERSION ${DPPP_GLOBAL_SOVERSIO
 
 # I'm avoiding cmake's test commands, because they build executable tests by default
 # which makes the compilation process more expensive.
-add_executable(runtests EXCLUDE_FROM_ALL
-  DPPP/test/runtests.cc
+set(TEST_FILENAMES
   DPPP/test/tMirror.cc
   DPPP/test/tPhaseShift.cc
+  DDECal/test/tRotationConstraint.cc
+)
+
+set(RUNTEST_SOURCES
+  DPPP/test/runtests.cc
+  ${TEST_FILENAMES}
   ${OS_SPECIFIC_TESTS}
   ${DPPP_OBJECT} ${COMMON_OBJECT} ${BLOB_OBJECT} ${DDECAL_OBJECT} ${IDGPREDICT_OBJECT} ${PARMDB_OBJECT} ${AOFLAGGERSTEP_OBJECT}
 )
+
+add_executable(runtests EXCLUDE_FROM_ALL ${RUNTEST_SOURCES})
+
 target_link_libraries(runtests ${EXTRA_LIBRARIES} ${LOFAR_STATION_RESPONSE_LIB})
+
+# Add each test as a cmake test (only for tests added with BOOST_AUTO_TEST_SUITE and BOOST_AUTO_TEST_CASE)
+foreach(TEST_FILENAME ${TEST_FILENAMES})
+  file(READ ${TEST_FILENAME} TEST_FILE_CONTENTS)
+
+  # Extract test suite name
+  string(REGEX MATCHALL "BOOST_AUTO_TEST_CASE\\( *([A-Za-z_0-9]+) *\\)" FOUND_TESTS ${TEST_FILE_CONTENTS})
+  string(REGEX MATCH "BOOST_AUTO_TEST_SUITE\\(*([A-Za-z_0-9]+)*\\)" TESTSUITE_NAME_FULL ${TEST_FILE_CONTENTS})
+
+  # Extract test name
+  string(REGEX REPLACE ".*\\( *([A-Za-z_0-9]+) *\\).*" "\\1" TESTSUITE_NAME ${TESTSUITE_NAME_FULL})
+  foreach(FOUND_TEST ${FOUND_TESTS})
+    string(REGEX REPLACE ".*\\( *([A-Za-z_0-9]+) *\\).*" "\\1" TEST_NAME ${FOUND_TEST})
+    add_test(NAME "${TESTSUITE_NAME}.${TEST_NAME}" COMMAND runtests --run_test=${TESTSUITE_NAME}/${TEST_NAME} --catch_system_error=yes)
```

Comment:
>     * `make test` doesn't work without `make check`. I tried to look up in the documentation of how to make the tests depend on runtest but can't find anything that works. A workaround is to put `runtests` in the ALL target (after which `make` followed by `make test` would work, but just `make test` still doesn't work, as apparently that's not supposed to work in cmake ( ? ) ), but I really dislike that, because I don't want to build the tests continuously when developing.

Should be fixed now.
 
>     * It adds mostly unnecessary complexity
>     * Searching for `BOOST_AUTO_TEST_CASE` etc in the source code is a bit ugly and limits the use of different Boost tests.

Adds complexity yes, but I hope these 12 lines of CMake code (though ugly) are documented and self-explanatory enough. Other kind of boost tests can be added later on by expanding the regexp.

>     * The advantage that I see of having the make-test output is small, if any (I like the output of 'make test', but it is not much different from `runtest -l unit_scope`, which executes faster -- but I like the concept of 'no output means no error' which an empty `runtest` would do a bit more, so you don't have to scroll back to see what happened).

This boils down to personal preference. In CTest, there's an option to show output of all failed tests; as well as the option `--quiet` to hide all output except when something fails.

> The only good reason that I see to use cmake testing, is if the more advanced test cases that use input data can not be covered without using cmake tests. Maybe we should investigate whether that's the case, and if they can be covered just keep it simple and only use Boost testing.

Since all current tests are based on CMake, I don't see a big advantage of rewriting everything into Boost. This hybrid approach enables you to write tests in boost, while it should be more straightforward to enable old cmake tests.